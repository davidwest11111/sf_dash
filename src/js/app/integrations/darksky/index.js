import axios from 'axios';
import jsonpAdapter from 'axios-jsonp';


export const URL_BASE = 'https://api.darksky.net/forecast';

export default class DarkSky {
  static getUrl = ({
    // all params required
    base = URL_BASE,
    key,
    lat,
    long,
    exclude = 'minutely,flags,currently',
    lang = 'en',
    units = 'auto',
  }) => `${base}/${key}/${lat},${long}?exclude=${exclude}&lang=${lang}&units=${units}`;

  static async fetchWeatherData({ lat, long, key, units }) {
    // NOTE: this is a jsonp call. there are no status codes or errors to parse like with XHR
    // try/catch errors in components that call this, so just return the response as-is
    const response = await axios.get(this.getUrl({ lat, long, key, units }), {
      adapter: jsonpAdapter, // gets around CORS
    });
    return response;
  }

  static async fetchAllLocationsWeatherData(locations, key, units) {
    const darkSkyUnits = this.convertUnitsForDarkSky(units);
    const responses = await axios.all(
      locations.map(({ lat, long }) => (
        this.fetchWeatherData({ lat, long, key, units: darkSkyUnits })
      )),
    );
    return responses;
  }

  static isValidKey = async key => {
    const response = await this.fetchWeatherData({ lat: 38, long: -122, key });
    return !!response?.data;
  }

  static convertUnitsForDarkSky = (units = 'auto') => {
    switch (units) {
      case ('f'):
        return 'us';
      case ('c'):
        return 'si';
      case ('auto'):
        return 'auto';
      default:
        throw new Error(`Invalid Unit type: ${units}`);
    }
  };
}
