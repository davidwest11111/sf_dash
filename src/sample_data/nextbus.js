// The data returned is actually in XML, this is just what the X2JS library outputs

export const SAMPLE_PREDICTION = {
  direction: {
    prediction: [
      {
        _epochTime: '1588198229803',
        _seconds: '296',
        _minutes: '4',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6609',
        _block: '1620',
        _tripTag: '9359239',
        $$hashKey: 'object:19',
      },
      {
        _epochTime: '1588198681179',
        _seconds: '747',
        _minutes: '12',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6599',
        _block: '1622',
        _tripTag: '9359240',
        $$hashKey: 'object:20',
      }, { _epochTime: '1588199412858', _seconds: '1479', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6547', _block: '1633', _tripTag: '9359237', $$hashKey: 'object:21' }, { _epochTime: '1588199419316', _seconds: '1486', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6533', _block: '1628', _tripTag: '9359243', $$hashKey: 'object:22' }],
    _title: 'Inbound to Ferry Plaza',
  },
  message: [
    {
      _text: 'Service frequency for this stop is 10 minutes or less.',
      _priority: 'Normal',
    },
    {
      _text: 'Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info.',
      _priority: 'Low',
    }, { _text: 'Text COVID19SF to 888-777 for official updates. Visit sfdph.org', _priority: 'Low' }, { _text: 'Board thru rear doors. Front door for accessibility needs only.', _priority: 'Normal' }],
  _agencyTitle: 'SF Muni',
  _routeTitle: '14-Mission',
  _routeTag: '14',
  _stopTitle: 'Mission St & 24th St',
  _stopTag: '5565',
  $$hashKey: 'object:17',
  className: '',
};

// FIXME
export const SAMPLE_FAILED_PREDICTIONS = {

};
