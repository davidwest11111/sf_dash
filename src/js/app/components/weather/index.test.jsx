import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import LocalStorage from 'app/utils/local_storage';
import DarkSky from 'app/integrations/darksky';
import { SAMPLE_LOCATION } from 'sample_data/darksky';

import Weather, { LOCATIONS, CALL_INTERVAL } from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let fetchAllLocationsWeatherDataSpy;

  beforeEach(() => {
    jest.spyOn(LocalStorage, 'get').mockReturnValue('foo');
    fetchAllLocationsWeatherDataSpy = jest.spyOn(DarkSky, 'fetchAllLocationsWeatherData')
      .mockReturnValue([]);
  });

  test('it initially loads', () => {
    wrapper = shallow((
      <Weather />
    ));
  });

  test('there are errors', () => {
    wrapper = shallow((
      <Weather />
    ));

    wrapper.setState({
      fetchError: true,
    });
  });

  test('state contains weather data', async () => {
    fetchAllLocationsWeatherDataSpy = jest.spyOn(DarkSky, 'fetchAllLocationsWeatherData')
      .mockReturnValue([{ data: { ...SAMPLE_LOCATION } }]);

    // updateWeatherLocations() gets called on componentDidMount
    wrapper = shallow((
      <Weather />
    ));

    expect(fetchAllLocationsWeatherDataSpy).toHaveBeenCalledWith(LOCATIONS, 'foo', 'foo');
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let wrapper;
  let instance;
  let setIntervalSpy;
  let clearIntervalSpy;

  beforeEach(() => {
    setIntervalSpy = jest.spyOn(window, 'setInterval').mockReturnValue(11);
    clearIntervalSpy = jest.spyOn(window, 'clearInterval');
    wrapper = shallow((
      <Weather />
    ));
    instance = wrapper.instance();
  });

  test('componentDidMount sets an interval', () => {
    expect(setIntervalSpy).toHaveBeenCalledWith(expect.any(Function), CALL_INTERVAL);
    expect(instance.interval).toEqual(11);
  });

  test('componentWillUnmount clears an interval', () => {
    wrapper.unmount();

    expect(clearIntervalSpy).toHaveBeenCalledWith(11);
  });
});
