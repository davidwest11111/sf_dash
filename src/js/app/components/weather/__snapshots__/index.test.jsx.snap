// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when has an errMsg 1`] = `
<div
  className="weather"
>
  <div
    className="error-msg"
  >
    <span
      className="fas fa-exclamation-circle mr-1"
    />
    <span>
      Weather request failed. Make sure 
      you've
       supplied an API Key in the
       
      <a
        href="#/settings"
      >
        settings
      </a>
      .
    </span>
  </div>
  <div
    className="d-flex justify-content-end"
  >
    <TimeSinceLastUpdated
      className=""
      lastUpdated={"2019-11-11T11:11:11.000Z"}
      loading={false}
      msUntilWarning={1800000}
    />
  </div>
</div>
`;

exports[`outputs the expected tree when it initially loads 1`] = `
<div
  className="weather"
>
  <div
    className="d-flex justify-content-end"
  >
    <TimeSinceLastUpdated
      className=""
      lastUpdated={"2019-11-11T11:11:11.000Z"}
      loading={false}
      msUntilWarning={1800000}
    />
  </div>
</div>
`;

exports[`outputs the expected tree when state contains weather data 1`] = `
<div
  className="weather"
>
  <Card
    className="mb-2"
    key="37.7448205,-122.4100494"
    tag="div"
  >
    <div
      className="location-readable"
    >
      <small>
        Bernal Heights
      </small>
    </div>
    <CardBody
      tag="div"
    >
      <Row
        className="slim-gutters"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="5"
        >
          <SidePanel
            alerts={
              Array [
                Object {
                  "description": "...FLOOD WATCH REMAINS IN EFFECT THROUGH LATE MONDAY NIGHT...
",
                  "expires": 1510036680,
                  "time": 1509993360,
                  "title": "Flood Watch for Mason, WA",
                  "uri": "http://alerts.weather.gov/cap/wwacapget.php?x=WA1255E4DB8494.FloodWatch.1255",
                },
              ]
            }
            className=""
            dailyForecasts={
              Array [
                Object {
                  "apparentTemperatureHigh": 15.95,
                  "apparentTemperatureHighTime": 1588285860,
                  "apparentTemperatureLow": 11.04,
                  "apparentTemperatureLowTime": 1588339020,
                  "apparentTemperatureMax": 15.95,
                  "apparentTemperatureMaxTime": 1588285860,
                  "apparentTemperatureMin": 11.49,
                  "apparentTemperatureMinTime": 1588251120,
                  "cloudCover": 0.41,
                  "dewPoint": 8.85,
                  "humidity": 0.75,
                  "icon": "partly-cloudy-day",
                  "moonPhase": 0.26,
                  "ozone": 317.9,
                  "precipIntensity": 0.0089,
                  "precipIntensityMax": 0.031,
                  "precipIntensityMaxTime": 1588309200,
                  "precipProbability": 0.4,
                  "precipType": "rain",
                  "pressure": 1019.5,
                  "summary": "Partly cloudy throughout the day.",
                  "sunriseTime": 1588252500,
                  "sunsetTime": 1588302060,
                  "temperatureHigh": 16.23,
                  "temperatureHighTime": 1588285860,
                  "temperatureLow": 10.77,
                  "temperatureLowTime": 1588339020,
                  "temperatureMax": 16.23,
                  "temperatureMaxTime": 1588285860,
                  "temperatureMin": 11.22,
                  "temperatureMinTime": 1588251120,
                  "time": 1588230000,
                  "uvIndex": 7,
                  "uvIndexTime": 1588278780,
                  "visibility": 16.093,
                  "windBearing": 266,
                  "windGust": 10.47,
                  "windGustTime": 1588295580,
                  "windSpeed": 4.22,
                },
                Object {
                  "apparentTemperatureHigh": 17.42,
                  "apparentTemperatureHighTime": 1588367220,
                  "apparentTemperatureLow": 12.85,
                  "apparentTemperatureLowTime": 1588430940,
                  "apparentTemperatureMax": 17.42,
                  "apparentTemperatureMaxTime": 1588367220,
                  "apparentTemperatureMin": 11.04,
                  "apparentTemperatureMinTime": 1588339020,
                  "cloudCover": 0.2,
                  "dewPoint": 6.98,
                  "humidity": 0.63,
                  "icon": "clear-day",
                  "moonPhase": 0.3,
                  "ozone": 343.3,
                  "precipIntensity": 0.0094,
                  "precipIntensityMax": 0.0454,
                  "precipIntensityMaxTime": 1588326900,
                  "precipProbability": 0.04,
                  "precipType": "rain",
                  "pressure": 1019.5,
                  "summary": "Clear throughout the day.",
                  "sunriseTime": 1588338840,
                  "sunsetTime": 1588388520,
                  "temperatureHigh": 17.7,
                  "temperatureHighTime": 1588367220,
                  "temperatureLow": 12.58,
                  "temperatureLowTime": 1588430940,
                  "temperatureMax": 17.7,
                  "temperatureMaxTime": 1588367220,
                  "temperatureMin": 10.77,
                  "temperatureMinTime": 1588339020,
                  "time": 1588316400,
                  "uvIndex": 8,
                  "uvIndexTime": 1588365360,
                  "visibility": 16.093,
                  "windBearing": 276,
                  "windGust": 9.14,
                  "windGustTime": 1588377660,
                  "windSpeed": 4.28,
                },
                Object {
                  "apparentTemperatureHigh": 15.9,
                  "apparentTemperatureHighTime": 1588458300,
                  "apparentTemperatureLow": 11.74,
                  "apparentTemperatureLowTime": 1588518000,
                  "apparentTemperatureMax": 15.9,
                  "apparentTemperatureMaxTime": 1588458300,
                  "apparentTemperatureMin": 12.85,
                  "apparentTemperatureMinTime": 1588430940,
                  "cloudCover": 0.87,
                  "dewPoint": 10.11,
                  "humidity": 0.77,
                  "icon": "cloudy",
                  "moonPhase": 0.33,
                  "ozone": 332.6,
                  "precipIntensity": 0.0188,
                  "precipIntensityMax": 0.0494,
                  "precipIntensityMaxTime": 1588418280,
                  "precipProbability": 0.1,
                  "precipType": "rain",
                  "pressure": 1020.3,
                  "summary": "Overcast throughout the day.",
                  "sunriseTime": 1588425180,
                  "sunsetTime": 1588474920,
                  "temperatureHigh": 16.18,
                  "temperatureHighTime": 1588458300,
                  "temperatureLow": 11.47,
                  "temperatureLowTime": 1588518000,
                  "temperatureMax": 16.18,
                  "temperatureMaxTime": 1588458300,
                  "temperatureMin": 12.58,
                  "temperatureMinTime": 1588430940,
                  "time": 1588402800,
                  "uvIndex": 5,
                  "uvIndexTime": 1588452300,
                  "visibility": 15.997,
                  "windBearing": 257,
                  "windGust": 6.37,
                  "windGustTime": 1588463040,
                  "windSpeed": 3.38,
                },
                Object {
                  "apparentTemperatureHigh": 15.01,
                  "apparentTemperatureHighTime": 1588540440,
                  "apparentTemperatureLow": 10.55,
                  "apparentTemperatureLowTime": 1588593780,
                  "apparentTemperatureMax": 15.01,
                  "apparentTemperatureMaxTime": 1588540440,
                  "apparentTemperatureMin": 11.7,
                  "apparentTemperatureMinTime": 1588519740,
                  "cloudCover": 0.44,
                  "dewPoint": 7.29,
                  "humidity": 0.68,
                  "icon": "partly-cloudy-day",
                  "moonPhase": 0.37,
                  "ozone": 327.1,
                  "precipIntensity": 0.0079,
                  "precipIntensityMax": 0.0203,
                  "precipIntensityMaxTime": 1588489200,
                  "precipProbability": 0.08,
                  "precipType": "rain",
                  "pressure": 1021.5,
                  "summary": "Partly cloudy throughout the day.",
                  "sunriseTime": 1588511520,
                  "sunsetTime": 1588561380,
                  "temperatureHigh": 15.29,
                  "temperatureHighTime": 1588540440,
                  "temperatureLow": 10.28,
                  "temperatureLowTime": 1588593780,
                  "temperatureMax": 15.29,
                  "temperatureMaxTime": 1588540440,
                  "temperatureMin": 11.43,
                  "temperatureMinTime": 1588519740,
                  "time": 1588489200,
                  "uvIndex": 8,
                  "uvIndexTime": 1588537860,
                  "visibility": 16.093,
                  "windBearing": 273,
                  "windGust": 11.61,
                  "windGustTime": 1588554480,
                  "windSpeed": 4.27,
                },
                Object {
                  "apparentTemperatureHigh": 17.9,
                  "apparentTemperatureHighTime": 1588633500,
                  "apparentTemperatureLow": 11.84,
                  "apparentTemperatureLowTime": 1588681860,
                  "apparentTemperatureMax": 17.9,
                  "apparentTemperatureMaxTime": 1588633500,
                  "apparentTemperatureMin": 10.55,
                  "apparentTemperatureMinTime": 1588593780,
                  "cloudCover": 0.01,
                  "dewPoint": 6.4,
                  "humidity": 0.6,
                  "icon": "clear-day",
                  "moonPhase": 0.41,
                  "ozone": 332.6,
                  "precipIntensity": 0.0016,
                  "precipIntensityMax": 0.011,
                  "precipIntensityMaxTime": 1588647000,
                  "precipProbability": 0.01,
                  "precipType": "rain",
                  "pressure": 1021.5,
                  "summary": "Clear throughout the day.",
                  "sunriseTime": 1588597860,
                  "sunsetTime": 1588647840,
                  "temperatureHigh": 18.18,
                  "temperatureHighTime": 1588633500,
                  "temperatureLow": 11.57,
                  "temperatureLowTime": 1588681860,
                  "temperatureMax": 18.18,
                  "temperatureMaxTime": 1588633500,
                  "temperatureMin": 10.28,
                  "temperatureMinTime": 1588593780,
                  "time": 1588575600,
                  "uvIndex": 9,
                  "uvIndexTime": 1588622700,
                  "visibility": 16.093,
                  "windBearing": 280,
                  "windGust": 8.53,
                  "windGustTime": 1588638060,
                  "windSpeed": 3.33,
                },
                Object {
                  "apparentTemperatureHigh": 16.77,
                  "apparentTemperatureHighTime": 1588719120,
                  "apparentTemperatureLow": 12.52,
                  "apparentTemperatureLowTime": 1588767480,
                  "apparentTemperatureMax": 16.77,
                  "apparentTemperatureMaxTime": 1588719120,
                  "apparentTemperatureMin": 11.84,
                  "apparentTemperatureMinTime": 1588681860,
                  "cloudCover": 0.35,
                  "dewPoint": 6.1,
                  "humidity": 0.58,
                  "icon": "partly-cloudy-day",
                  "moonPhase": 0.45,
                  "ozone": 329,
                  "precipIntensity": 0.0036,
                  "precipIntensityMax": 0.0072,
                  "precipIntensityMaxTime": 1588744800,
                  "precipProbability": 0.05,
                  "precipType": "rain",
                  "pressure": 1021.6,
                  "summary": "Partly cloudy throughout the day.",
                  "sunriseTime": 1588684200,
                  "sunsetTime": 1588734300,
                  "temperatureHigh": 17.05,
                  "temperatureHighTime": 1588719120,
                  "temperatureLow": 12.25,
                  "temperatureLowTime": 1588767480,
                  "temperatureMax": 17.05,
                  "temperatureMaxTime": 1588719120,
                  "temperatureMin": 11.57,
                  "temperatureMinTime": 1588681860,
                  "time": 1588662000,
                  "uvIndex": 5,
                  "uvIndexTime": 1588705380,
                  "visibility": 16.093,
                  "windBearing": 292,
                  "windGust": 10.83,
                  "windGustTime": 1588731060,
                  "windSpeed": 4.22,
                },
              ]
            }
            hourlyForecasts={
              Array [
                Object {
                  "apparentTemperature": 15.49,
                  "cloudCover": 0.36,
                  "dewPoint": 10.85,
                  "humidity": 0.74,
                  "icon": "partly-cloudy-day",
                  "ozone": 319.6,
                  "precipIntensity": 0,
                  "precipProbability": 0.4,
                  "pressure": 1019.9,
                  "summary": "Partly Cloudy",
                  "temperature": 15.49,
                  "time": 1588276800,
                  "uvIndex": 7,
                  "visibility": 16.093,
                  "windBearing": 263,
                  "windGust": 5.76,
                  "windSpeed": 4.44,
                },
                Object {
                  "apparentTemperature": 15.71,
                  "cloudCover": 0.29,
                  "dewPoint": 10.41,
                  "humidity": 0.71,
                  "icon": "clear-day",
                  "ozone": 319.9,
                  "precipIntensity": 0,
                  "precipProbability": 0,
                  "pressure": 1019.7,
                  "summary": "Clear",
                  "temperature": 15.71,
                  "time": 1588280400,
                  "uvIndex": 7,
                  "visibility": 16.093,
                  "windBearing": 262,
                  "windGust": 7.32,
                  "windSpeed": 5.04,
                },
                Object {
                  "apparentTemperature": 15.91,
                  "cloudCover": 0.35,
                  "dewPoint": 9.74,
                  "humidity": 0.67,
                  "icon": "partly-cloudy-day",
                  "ozone": 320,
                  "precipIntensity": 0,
                  "precipProbability": 0,
                  "pressure": 1019.4,
                  "summary": "Partly Cloudy",
                  "temperature": 15.91,
                  "time": 1588284000,
                  "uvIndex": 6,
                  "visibility": 16.093,
                  "windBearing": 259,
                  "windGust": 8.8,
                  "windSpeed": 5.61,
                },
                Object {
                  "apparentTemperature": 15.87,
                  "cloudCover": 0.27,
                  "dewPoint": 8.7,
                  "humidity": 0.62,
                  "icon": "clear-day",
                  "ozone": 320,
                  "precipIntensity": 0.0051,
                  "precipProbability": 0.01,
                  "precipType": "rain",
                  "pressure": 1019.6,
                  "summary": "Clear",
                  "temperature": 15.87,
                  "time": 1588287600,
                  "uvIndex": 4,
                  "visibility": 16.093,
                  "windBearing": 280,
                  "windGust": 9.54,
                  "windSpeed": 6.2,
                },
              ]
            }
          />
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="7"
        >
          <HourlyForecast
            className=""
            hourlyForecasts={
              Array [
                Object {
                  "apparentTemperature": 15.49,
                  "cloudCover": 0.36,
                  "dewPoint": 10.85,
                  "humidity": 0.74,
                  "icon": "partly-cloudy-day",
                  "ozone": 319.6,
                  "precipIntensity": 0,
                  "precipProbability": 0.4,
                  "pressure": 1019.9,
                  "summary": "Partly Cloudy",
                  "temperature": 15.49,
                  "time": 1588276800,
                  "uvIndex": 7,
                  "visibility": 16.093,
                  "windBearing": 263,
                  "windGust": 5.76,
                  "windSpeed": 4.44,
                },
                Object {
                  "apparentTemperature": 15.71,
                  "cloudCover": 0.29,
                  "dewPoint": 10.41,
                  "humidity": 0.71,
                  "icon": "clear-day",
                  "ozone": 319.9,
                  "precipIntensity": 0,
                  "precipProbability": 0,
                  "pressure": 1019.7,
                  "summary": "Clear",
                  "temperature": 15.71,
                  "time": 1588280400,
                  "uvIndex": 7,
                  "visibility": 16.093,
                  "windBearing": 262,
                  "windGust": 7.32,
                  "windSpeed": 5.04,
                },
                Object {
                  "apparentTemperature": 15.91,
                  "cloudCover": 0.35,
                  "dewPoint": 9.74,
                  "humidity": 0.67,
                  "icon": "partly-cloudy-day",
                  "ozone": 320,
                  "precipIntensity": 0,
                  "precipProbability": 0,
                  "pressure": 1019.4,
                  "summary": "Partly Cloudy",
                  "temperature": 15.91,
                  "time": 1588284000,
                  "uvIndex": 6,
                  "visibility": 16.093,
                  "windBearing": 259,
                  "windGust": 8.8,
                  "windSpeed": 5.61,
                },
                Object {
                  "apparentTemperature": 15.87,
                  "cloudCover": 0.27,
                  "dewPoint": 8.7,
                  "humidity": 0.62,
                  "icon": "clear-day",
                  "ozone": 320,
                  "precipIntensity": 0.0051,
                  "precipProbability": 0.01,
                  "precipType": "rain",
                  "pressure": 1019.6,
                  "summary": "Clear",
                  "temperature": 15.87,
                  "time": 1588287600,
                  "uvIndex": 4,
                  "visibility": 16.093,
                  "windBearing": 280,
                  "windGust": 9.54,
                  "windSpeed": 6.2,
                },
              ]
            }
          />
        </Col>
      </Row>
    </CardBody>
    <CardFooter
      className="bg-transparent py-0"
      tag="div"
    >
      <ThreeDayForecast
        className=""
        dailyForecasts={
          Array [
            Object {
              "apparentTemperatureHigh": 15.95,
              "apparentTemperatureHighTime": 1588285860,
              "apparentTemperatureLow": 11.04,
              "apparentTemperatureLowTime": 1588339020,
              "apparentTemperatureMax": 15.95,
              "apparentTemperatureMaxTime": 1588285860,
              "apparentTemperatureMin": 11.49,
              "apparentTemperatureMinTime": 1588251120,
              "cloudCover": 0.41,
              "dewPoint": 8.85,
              "humidity": 0.75,
              "icon": "partly-cloudy-day",
              "moonPhase": 0.26,
              "ozone": 317.9,
              "precipIntensity": 0.0089,
              "precipIntensityMax": 0.031,
              "precipIntensityMaxTime": 1588309200,
              "precipProbability": 0.4,
              "precipType": "rain",
              "pressure": 1019.5,
              "summary": "Partly cloudy throughout the day.",
              "sunriseTime": 1588252500,
              "sunsetTime": 1588302060,
              "temperatureHigh": 16.23,
              "temperatureHighTime": 1588285860,
              "temperatureLow": 10.77,
              "temperatureLowTime": 1588339020,
              "temperatureMax": 16.23,
              "temperatureMaxTime": 1588285860,
              "temperatureMin": 11.22,
              "temperatureMinTime": 1588251120,
              "time": 1588230000,
              "uvIndex": 7,
              "uvIndexTime": 1588278780,
              "visibility": 16.093,
              "windBearing": 266,
              "windGust": 10.47,
              "windGustTime": 1588295580,
              "windSpeed": 4.22,
            },
            Object {
              "apparentTemperatureHigh": 17.42,
              "apparentTemperatureHighTime": 1588367220,
              "apparentTemperatureLow": 12.85,
              "apparentTemperatureLowTime": 1588430940,
              "apparentTemperatureMax": 17.42,
              "apparentTemperatureMaxTime": 1588367220,
              "apparentTemperatureMin": 11.04,
              "apparentTemperatureMinTime": 1588339020,
              "cloudCover": 0.2,
              "dewPoint": 6.98,
              "humidity": 0.63,
              "icon": "clear-day",
              "moonPhase": 0.3,
              "ozone": 343.3,
              "precipIntensity": 0.0094,
              "precipIntensityMax": 0.0454,
              "precipIntensityMaxTime": 1588326900,
              "precipProbability": 0.04,
              "precipType": "rain",
              "pressure": 1019.5,
              "summary": "Clear throughout the day.",
              "sunriseTime": 1588338840,
              "sunsetTime": 1588388520,
              "temperatureHigh": 17.7,
              "temperatureHighTime": 1588367220,
              "temperatureLow": 12.58,
              "temperatureLowTime": 1588430940,
              "temperatureMax": 17.7,
              "temperatureMaxTime": 1588367220,
              "temperatureMin": 10.77,
              "temperatureMinTime": 1588339020,
              "time": 1588316400,
              "uvIndex": 8,
              "uvIndexTime": 1588365360,
              "visibility": 16.093,
              "windBearing": 276,
              "windGust": 9.14,
              "windGustTime": 1588377660,
              "windSpeed": 4.28,
            },
            Object {
              "apparentTemperatureHigh": 15.9,
              "apparentTemperatureHighTime": 1588458300,
              "apparentTemperatureLow": 11.74,
              "apparentTemperatureLowTime": 1588518000,
              "apparentTemperatureMax": 15.9,
              "apparentTemperatureMaxTime": 1588458300,
              "apparentTemperatureMin": 12.85,
              "apparentTemperatureMinTime": 1588430940,
              "cloudCover": 0.87,
              "dewPoint": 10.11,
              "humidity": 0.77,
              "icon": "cloudy",
              "moonPhase": 0.33,
              "ozone": 332.6,
              "precipIntensity": 0.0188,
              "precipIntensityMax": 0.0494,
              "precipIntensityMaxTime": 1588418280,
              "precipProbability": 0.1,
              "precipType": "rain",
              "pressure": 1020.3,
              "summary": "Overcast throughout the day.",
              "sunriseTime": 1588425180,
              "sunsetTime": 1588474920,
              "temperatureHigh": 16.18,
              "temperatureHighTime": 1588458300,
              "temperatureLow": 11.47,
              "temperatureLowTime": 1588518000,
              "temperatureMax": 16.18,
              "temperatureMaxTime": 1588458300,
              "temperatureMin": 12.58,
              "temperatureMinTime": 1588430940,
              "time": 1588402800,
              "uvIndex": 5,
              "uvIndexTime": 1588452300,
              "visibility": 15.997,
              "windBearing": 257,
              "windGust": 6.37,
              "windGustTime": 1588463040,
              "windSpeed": 3.38,
            },
            Object {
              "apparentTemperatureHigh": 15.01,
              "apparentTemperatureHighTime": 1588540440,
              "apparentTemperatureLow": 10.55,
              "apparentTemperatureLowTime": 1588593780,
              "apparentTemperatureMax": 15.01,
              "apparentTemperatureMaxTime": 1588540440,
              "apparentTemperatureMin": 11.7,
              "apparentTemperatureMinTime": 1588519740,
              "cloudCover": 0.44,
              "dewPoint": 7.29,
              "humidity": 0.68,
              "icon": "partly-cloudy-day",
              "moonPhase": 0.37,
              "ozone": 327.1,
              "precipIntensity": 0.0079,
              "precipIntensityMax": 0.0203,
              "precipIntensityMaxTime": 1588489200,
              "precipProbability": 0.08,
              "precipType": "rain",
              "pressure": 1021.5,
              "summary": "Partly cloudy throughout the day.",
              "sunriseTime": 1588511520,
              "sunsetTime": 1588561380,
              "temperatureHigh": 15.29,
              "temperatureHighTime": 1588540440,
              "temperatureLow": 10.28,
              "temperatureLowTime": 1588593780,
              "temperatureMax": 15.29,
              "temperatureMaxTime": 1588540440,
              "temperatureMin": 11.43,
              "temperatureMinTime": 1588519740,
              "time": 1588489200,
              "uvIndex": 8,
              "uvIndexTime": 1588537860,
              "visibility": 16.093,
              "windBearing": 273,
              "windGust": 11.61,
              "windGustTime": 1588554480,
              "windSpeed": 4.27,
            },
            Object {
              "apparentTemperatureHigh": 17.9,
              "apparentTemperatureHighTime": 1588633500,
              "apparentTemperatureLow": 11.84,
              "apparentTemperatureLowTime": 1588681860,
              "apparentTemperatureMax": 17.9,
              "apparentTemperatureMaxTime": 1588633500,
              "apparentTemperatureMin": 10.55,
              "apparentTemperatureMinTime": 1588593780,
              "cloudCover": 0.01,
              "dewPoint": 6.4,
              "humidity": 0.6,
              "icon": "clear-day",
              "moonPhase": 0.41,
              "ozone": 332.6,
              "precipIntensity": 0.0016,
              "precipIntensityMax": 0.011,
              "precipIntensityMaxTime": 1588647000,
              "precipProbability": 0.01,
              "precipType": "rain",
              "pressure": 1021.5,
              "summary": "Clear throughout the day.",
              "sunriseTime": 1588597860,
              "sunsetTime": 1588647840,
              "temperatureHigh": 18.18,
              "temperatureHighTime": 1588633500,
              "temperatureLow": 11.57,
              "temperatureLowTime": 1588681860,
              "temperatureMax": 18.18,
              "temperatureMaxTime": 1588633500,
              "temperatureMin": 10.28,
              "temperatureMinTime": 1588593780,
              "time": 1588575600,
              "uvIndex": 9,
              "uvIndexTime": 1588622700,
              "visibility": 16.093,
              "windBearing": 280,
              "windGust": 8.53,
              "windGustTime": 1588638060,
              "windSpeed": 3.33,
            },
            Object {
              "apparentTemperatureHigh": 16.77,
              "apparentTemperatureHighTime": 1588719120,
              "apparentTemperatureLow": 12.52,
              "apparentTemperatureLowTime": 1588767480,
              "apparentTemperatureMax": 16.77,
              "apparentTemperatureMaxTime": 1588719120,
              "apparentTemperatureMin": 11.84,
              "apparentTemperatureMinTime": 1588681860,
              "cloudCover": 0.35,
              "dewPoint": 6.1,
              "humidity": 0.58,
              "icon": "partly-cloudy-day",
              "moonPhase": 0.45,
              "ozone": 329,
              "precipIntensity": 0.0036,
              "precipIntensityMax": 0.0072,
              "precipIntensityMaxTime": 1588744800,
              "precipProbability": 0.05,
              "precipType": "rain",
              "pressure": 1021.6,
              "summary": "Partly cloudy throughout the day.",
              "sunriseTime": 1588684200,
              "sunsetTime": 1588734300,
              "temperatureHigh": 17.05,
              "temperatureHighTime": 1588719120,
              "temperatureLow": 12.25,
              "temperatureLowTime": 1588767480,
              "temperatureMax": 17.05,
              "temperatureMaxTime": 1588719120,
              "temperatureMin": 11.57,
              "temperatureMinTime": 1588681860,
              "time": 1588662000,
              "uvIndex": 5,
              "uvIndexTime": 1588705380,
              "visibility": 16.093,
              "windBearing": 292,
              "windGust": 10.83,
              "windGustTime": 1588731060,
              "windSpeed": 4.22,
            },
          ]
        }
      />
    </CardFooter>
  </Card>
  <div
    className="d-flex justify-content-end"
  >
    <TimeSinceLastUpdated
      className=""
      lastUpdated={"2019-11-11T11:11:11.000Z"}
      loading={false}
      msUntilWarning={1800000}
    />
  </div>
</div>
`;

exports[`outputs the expected tree when there are errors 1`] = `
<div
  className="weather"
>
  <div
    className="error-msg"
  >
    <span
      className="fas fa-exclamation-circle mr-1"
    />
    <span>
      Weather request failed. Make sure 
      you've
       supplied an API Key in the
       
      <a
        href="#/settings"
      >
        settings
      </a>
      .
    </span>
  </div>
  <div
    className="d-flex justify-content-end"
  >
    <TimeSinceLastUpdated
      className=""
      lastUpdated={"2019-11-11T11:11:11.000Z"}
      loading={false}
      msUntilWarning={1800000}
    />
  </div>
</div>
`;
